### PR TITLE
[AC-1754] Provide upgrade flow for paid organizations

### PR DESCRIPTION
**Pull request by amorask-bitwarden**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

[AC-1754](https://bitwarden.atlassian.net/browse/AC-1754)

## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

[AC-1754]: https://bitwarden.atlassian.net/browse/AC-1754?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by cturnbull-bitwarden — DISMISSED** [review]

**Comment by eliykat** [comment]
Removing AC Team as a reviewer because we're not a codeowner and I assume we were just pulled in accidentally. If this isn't the case, please re-request a review and we'd be happy to look at it.

**Review by cyprain-okeke — DISMISSED** [review]
Looks Very good to me

**Review by cyprain-okeke — APPROVED** [review]

**Review by cyprain-okeke — APPROVED** [review]

**Comment by amorask-bitwarden** [comment]
Is it possible to resole this comment as part of this PR as well since I think they're basically the same issue? https://bitwarden.atlassian.net/browse/AC-1987?focusedCommentId=48669

File: apps/web/src/app/billing/organizations/organization-plans.component.ts [hunk 1]
```diff
@@ -386,25 +400,52 @@ export class OrganizationPlansComponent implements OnInit, OnDestroy {
   }
 
   changedProduct() {
-    this.formGroup.controls.plan.setValue(this.selectablePlans[0].type);
+    const selectedPlan = this.selectablePlans[0];
```
Comment (cyprain-okeke) [inline apps/web/src/app/billing/organizations/organization-plans.component.ts:403]: This method seems to be getting longer, can we possibly move the if statements to smaller method base on how they are related

**Review by cyprain-okeke — APPROVED** [review]
Looks Good. Just added one comment

File: apps/web/src/app/billing/organizations/organization-plans.component.ts (hunk 1 above)
Comment (cturnbull-bitwarden) [reply apps/web/src/app/billing/organizations/organization-plans.component.ts:403]: Definitely, I'll push an update shortly

**Review by cyprain-okeke — APPROVED** [review]

**Review by r-tome — APPROVED** [review]

**Pull request merged**